linera-execution/tests/test_execution.rs

```diff
@@ -1667,3 +1660,90 @@ async fn test_close_chain() {
     .unwrap();
     assert!(view.system.closed.get());
 }
+
+/// Tests if an application can't transfer the tokens in the chain's balance while executing
+/// messages.
+#[tokio::test]
+async fn test_message_handler_cant_spend_chain_balance() -> anyhow::Result<()> {
```

Comment:
Do we have a test for the "happy path"? Also, it looks like you've added two match cases that fail with unauthenticated owner errors but we have a test for one, right?

Reply:
I added some more test cases, but they still don't cover the first `match` arm, which is equivalent to the previous `if` condition, because that's out of this PR's scope (access to the balance shared by all chain owners). I opened issue #1852 to track that though.